ChiTech/ChiPhysics/PhysicsMaterial/transportxsections/prop_transportxs_06_export.cc

```diff
@@ -0,0 +1,187 @@
+
+  std::ofstream ofile(file_name);
+
+  //======================================== Writing header info
+  ofile << "# Exported cross section from ChiTech\n";
+  ofile << "# Date: " << ChiTimer::GetLocalDateTimeString() << "\n";
+  ofile << "NUM_GROUPS " << num_groups << "\n";
+  ofile << "NUM_MOMENTS " << scattering_order+1 << "\n";
+  if (num_precursors>0)
+    ofile << "NUM_PRECURSORS " << num_precursors << "\n";
+
+  //======================================== Sigma_t
+  ofile << "\n";
+  ofile << "SIGMA_T_BEGIN\n";
+  {
+    int g=0;
+    for (auto val : sigma_tg)
+      ofile << g++ << " " << val << "\n";
+  }
+  ofile << "SIGMA_T_END\n";
+
+  //======================================== Sigma_f
+  bool sigma_f_filled = false;
+  for (auto val : sigma_fg)
+    if (val > 1.0e-20) sigma_f_filled = true;
+
+  if (sigma_f_filled)
+  {
+    ofile << "\n";
+    ofile << "SIGMA_F_BEGIN\n";
+    {
+      int g=0;
+      for (auto val : sigma_tg)
+        ofile << g++ << " " << val << "\n";
+    }
+    ofile << "SIGMA_F_END\n";
+  }
+
+  //======================================== Nu
+  bool nu_filled = false;
+  for (auto val : nu_sigma_fg)
+    if (val > 1.0e-20) nu_filled = true;
+
+  if (nu_filled)
+  {
+    ofile << "\n";
+    ofile << "NU_BEGIN\n";
+    {
+      int g=0;
+      for (auto val : nu_sigma_fg)
+      {
+        int gval = g++;
+        ofile << gval << " " << val/sigma_fg[g-1] << "\n";
+      }
+    }
+    ofile << "NU_END\n";
+  }
+
+  //======================================== Nu-prompt
+  bool nu_prompt_filled = false;
+  for (auto val : nu_p_sigma_fg)
+    if (val > 1.0e-20) nu_prompt_filled = true;
+
+  if (nu_prompt_filled)
+  {
+    ofile << "\n";
+    ofile << "NU_PROMPT_BEGIN\n";
+    {
+      int g=0;
+      for (auto val : nu_p_sigma_fg)
+      {
+        int gval = g++;
+        ofile << gval << " " << val/sigma_fg[g-1] << "\n";
+      }
+    }
+    ofile << "NU_PROMPT_END\n";
+  }
+
+  //======================================== Nu-delayed
+  bool nu_delayed_filled = false;
+  for (auto val : nu_d_sigma_fg)
+    if (val > 1.0e-20) nu_delayed_filled = true;
+
+  if (nu_delayed_filled)
+  {
+    ofile << "\n";
+    ofile << "NU_DELAYED_BEGIN\n";
+    {
+      int g=0;
+      for (auto val : nu_d_sigma_fg)
+      {
+        int gval = g++;
+        ofile << gval << " " << val/sigma_fg[g-1] << "\n";
+      }
+    }
+    ofile << "NU_DELAYED_END\n";
+  }
+
+  //======================================== Chi-prompt
+  bool chi_prompt_filled = false;
+  for (auto val : chi_g)
+    if (val > 1.0e-20) chi_prompt_filled = true;
+
+  if (chi_prompt_filled)
+  {
+    ofile << "\n";
+    ofile << "CHI_PROMPT_BEGIN\n";
+    {
+      int g=0;
+      for (auto val : chi_g)
+      {
+        int gval = g++;
+        ofile << gval << " " << val << "\n";
+      }
+    }
+    ofile << "CHI_PROMPT_END\n";
+  }
+
+  //======================================== Chi-delayed
+
```

Comment:
Is the export of chi-delayed omitted intentionally?

Reply:
Yes kind-of... But now I included it. Thanks! Good catch.